app/_kong_plugins/openid-connect/index.md

```diff
@@ -37,6 +35,541 @@ search_aliases:
+    httpbin->>kong: response
+    deactivate httpbin
+    activate client
+    kong->>client: response
+    deactivate kong
+    deactivate client
+{% endmermaid %}
+<!--vale on-->
+
+### Kong OAuth token auth flow
+
+The OpenID Connect plugin can verify the tokens issued by the [OAuth 2.0 plugin](/plugins/oauth2/).
+This is very similar to third party identity provider issued [JWT access token authentication](#jwt-access-token-authentication-flow)
+or [introspection authentication](#introspection-authentication-flow):
+
+<!--vale off-->
+{% mermaid %}
+sequenceDiagram
+    autonumber
+    participant client as Client <br>(e.g. mobile app)
+    participant kong as API Gateway <br>(Kong)
+    participant httpbin as Upstream <br>(upstream service,<br> e.g. httpbin)
+    activate client
+    activate kong
+    client->>kong: service with<br>access token
+    deactivate client
+    kong->>kong: load access token
+    kong->>kong: verify kong<br>oauth token
+    activate httpbin
+    kong->>httpbin: request with<br>access token
+    httpbin->>kong: response
+    deactivate httpbin
+    activate client
+    kong->>client: response
+    deactivate kong
+    deactivate client
+{% endmermaid %}
+<!--vale on-->
+
+
+### Password grant workflow
+
+Password grant is a **legacy** authentication grant. This is a less secure way of
+authenticating end users than the authorization code flow, because, for example,
+the passwords are shared with third parties.
+
+<!--vale off-->
+{% mermaid %}
+sequenceDiagram
+    autonumber
+    participant client as Client <br>(e.g. mobile app)
+    participant kong as API Gateway <br>(Kong)
+    participant idp as IdP <br>(e.g. Keycloak)
+    participant httpbin as Upstream <br>(upstream service,<br> e.g. httpbin)
+    activate client
+    activate kong
+    client->>kong: service with<br>basic authentication
+    deactivate client
+    kong->>kong: load <br>basic authentication<br>credentials
+    activate idp
+    kong->>idp: IdP/token with<br>client credentials and<br>password grant
+    deactivate kong
+    idp->>idp: authenticate client and<br>verify password grant
+    activate kong
+    idp->>kong: return tokens
+    deactivate idp
+    kong->>kong: verify tokens
+    activate httpbin
+    kong->>httpbin: request with access token
+    httpbin->>kong: response
+    deactivate httpbin
+    activate client
+    kong->>client: response
+    deactivate kong
+    deactivate client
+{% endmermaid %}
+<!--vale on-->
+
+### Refresh token grant workflow
+
+The refresh token grant can be used when the client has a refresh token available. There is a caveat
+with this: in general, identity providers only allow the refresh token grant to be executed with the same
+client that originally got the refresh token, and if there is a mismatch, it may not work. The mismatch
+is likely when the OpenID Connect plugin is configured to use one client, and the refresh token is retrieved
+with another. 
```

Comment:
```suggestion
The refresh token grant can be used when the client has a refresh token available. There is a caveat with this: in general, identity providers only allow the refresh token grant to be executed with the same client that originally got the refresh token, and if there is a mismatch, it may not work. The mismatch is likely when the OpenID Connect plugin is configured to use one client, and the refresh token is retrieved with another. 
```